.github/workflows/ci-main.yml

```diff
@@ -332,106 +332,99 @@ jobs:
+          - os: darwin
+            profile: namespace-profile-darwin # profile created in namespace console
+    name: Bazel Test arm64-${{ matrix.os }}
+    runs-on: ${{ matrix.profile }}
+    timeout-minutes: 120
     if: github.repository == 'dfinity/ic' # only run on public repo, not private since Namespace runners are not configured there, so these CI jobs get stuck otherwise.
     steps:
       - name: Set up Bazel cache
         run: |
           # Creates a bazelrc configuration fragment which tells bazel where the cache lives.
           nsc bazel cache setup --bazelrc=/tmp/bazel-cache.bazelrc
+
       - uses: actions/checkout@v4
+
+      - name: Set up zig cache
+        run: |
+          mkdir -p /tmp/zig-cache
+
       # We need to use lld on aarch64-linux because the default gold linker runs into the following linker bug:
       # /usr/bin/ld.gold: internal error in update_erratum_insn, at ../../gold/aarch64.cc:1003
       - name: Install lld
+        if: ${{ matrix.os == 'linux' }}
         run: |
           sudo apt install lld
 
-      - name: Build and Test
+      - name: Build PocketIC server to publish
         run: |
-          # run pocket-ic tests
           bazel \
             --noworkspace_rc \
             --bazelrc=./bazel/conf/.bazelrc.build --bazelrc=/tmp/bazel-cache.bazelrc \
-            test \
-              //packages/pocket-ic:all \
-              //rs/pocket_ic_server:test \
-              //rs/pocket_ic_server:gateway \
-              //rs/pocket_ic_server:pocket-ic-server \
-              //publish/binaries:pocket-ic.gz
+            build \
+            //publish/binaries:pocket-ic.gz
 
           mkdir -p build
           cp \
             ./bazel-bin/publish/binaries/pocket-ic.gz \
-            ./build/pocket-ic-server-arm64-linux.gz
-
-      - name: Upload pocket-ic-server
-        uses: actions/upload-artifact@v4
-        with:
-          name: pocket-ic-server-arm64-linux.gz
-          path: ./build/pocket-ic-server-arm64-linux.gz
+            ./build/pocket-ic-server-arm64-${{ matrix.os }}.gz
 
-  bazel-test-arm64-darwin:
-    name: Bazel Test macOS Apple Silicon
-    timeout-minutes: 120
-    runs-on: namespace-profile-darwin # profile created in namespace console
-    if: github.repository == 'dfinity/ic' # only run on public repo, not private since Namespace runners are not configured there, so these CI jobs get stuck otherwise.
-    steps:
-      - name: Set up Bazel cache
+      # Ensures that local development works on arm64-darwin.
+      - name: Build the entire codebase on darwin
+        if: ${{ matrix.os == 'darwin' }}
         run: |
-          # Creates a bazelrc configuration fragment which tells bazel where the cache lives.
-          nsc bazel cache setup --bazelrc=/tmp/bazel-cache.bazelrc
-      - uses: actions/checkout@v4
+          # We do not use `/...` here to exclude `//publish/...` which is not required for local development.
+          bazel_build_targets=(
```

Comment:
I think we can inline those in the command for simplicity, or am I missing something?